### DISCUSSION TITLE
Proposal: automatically close issues which don't use the issue template provided

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#10657] [Suggest an Idea]
#### What's the problem right now?

I'm getting tired of asking people to pretty please use the issue template.
To clarify my problem, here is what a user needs to do to end up with no template.

Sequence of events:

1. User wants to report bug.
2. Goes to Renovate bot repository.
3. Clicks issues tab,
4. Clicks "new issue".
5. Select bug report -> Get started.
6. Open new screen.
7. Screen contains template.
8. Deliberately throws away template and writes own text.

Note that we don't allow any empty issues via this setting:

https://github.com/renovatebot/renovate/blob/84f7fb0407411725c612a46260c0d2a3c9cf039e/.github/ISSUE_TEMPLATE/config.yml#L1

So the only way to end up with no template is by users deleting it themselves.

#### Why I don't like this behavior:

I find it rude to throw away the template. It basically says: "I don't want to follow this repository's established process, but still want the maintainers to spend time debugging/fixing the bug."

#### Proposed solution

Add new label: `bad-template` or something like that, and use the `label-actions.yml` file to get our bot to close the issue with an appropriate message like:

```markdown
We've closed this issue, as you're not using the proper template. 
Please open a new issue with the proper template.
```

#### Discuss first before making changes:

Before we make such a policy change, we should discuss it. 😉 

@viceice, @rarkins, @JamieMagee what do you think of my idea to close issues that don't use the proper template?

**Comment by rarkins** [comment]
I'm mostly with you but I wondered if possibly a tool like GitHub for Windows is partly responsible for discarding the template?

Also wondering if we can give people a day somehow to fix it before autoclosing.

**Reply by HonkingGoose** [reply]
> I wondered if possibly a tool like GitHub for Windows is partly responsible for discarding the template?

GitHub Desktop only allows you to open a pull request, and that uses the proper PR template.

GitHub's CLI tool `gh` allows you to create a issue from the command line. From reading the docs, it appears that it does not use the template. https://cli.github.com/manual/gh_issue_create

So maybe it's not the case that users are doing this deliberately - at least not in all cases - as it might be an honest mistake. So our automation should give a "grace period" to correct the mistake before taking action.

---

> Also wondering if we can give people a day somehow to fix it before autoclosing.

I don't think we can use the current `stale-action.yml`, as that is already setup for managing issues without reproduction, with the messages we need for that workflow.
Maybe copy/pasting and adjusting that file for a `stale-action-for-issue-without-template.yml` would work though.

**Reply by viceice** [reply]
Sounds good

**Reply by rarkins** [reply]
I'm happy to close these within 24-48 hours if the template isn't used. But it may need us to remove the label instead of it being fully automated

**Comment by HonkingGoose** [comment]
EDIT: We're now done with our review of the GitHub Actions we have now.

---

We're reviewing the GitHub Actions in use right now (see #10794), so I'll hold of the search until we're done with that.

When we're done reviewing the current actions it will be clearer what kind of actions we accept and reject. This will make searching for a suitable action easier.

**Comment by HonkingGoose** [comment]
I'm working on new Issue _Forms_ for the Renovate bot repository now (see issue #10860). It's basically like any form you see on the web, with required fields/inputs. The end-goal with _Forms_ is that its now way harder to mangle the required fields by accident.

People can still mess up the form afterwards though, by editing the raw Markdown they end up with after clicking on "Create Issue"... So it doesn't completely fix the problem of bad user input, but it should reduce it noticeably.

Not sure if the "automatically close issues which don't use the template"-idea is still something we want to pursue now that _Forms_ is on the horizon?

**Reply by viceice** [reply]
Let's defer any decisions until  we've got the new forms live. 🤗

**Reply by rarkins** [reply]
FYI I've seen that "convert discussion to issue" functionality bypasses the templates too. Any of these bypasses via github's web UI are unfortunate because means we can't enforce it so reliably

**Reply by HonkingGoose** [reply]
I went searching for all the ways to bypass the templates, and there are _way_ more than I expected... 😄 

Relevant quotes from [GitHub docs, creating an issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-an-issue):

> People with read permissions can create an issue in a repository where issues are enabled.

> From the right sidebar:
> - Creating an issue from a repository
> - Creating an issue from a comment
> - Creating an issue from code
> - Creating an issue from a project board note
> - Creating an issue from a task list item
> - Creating an issue from a URL query
> - Creating an issue from a discussion (not listed on GitHub docs for some reason!)

I think the main problem is that people with _read_ permissions have all those ways to get around the templates. It would be nicer if only people with some kind of permissions on the repo (_triage/write/admin/owner_) could get around the templates/forms. That would mean that the people who can get around the templates also have the rights to edit the top post to confirm to the templates/forms.

**Reply by rarkins** [reply]
Although I want to avoid people rudely/lazily deleting templates, I prefer even more to avoid discouraging contributors who've somehow ended up with a template-less issue through no fault of their own. So maybe we continue with manually asking people and hope that the new template capabilities reduce the occurrences

**Reply by HonkingGoose** [reply]
Okay, let's keep things as-is for now, and monitor if we maybe later do want/need a bot/action.
